radix.go

```diff
@@ -275,18 +275,19 @@ DELETE:
 	n.leaf = nil
 	t.size--
 
+	// Check if we should delete this node from the parent
+	if parent != nil && len(n.edges) == 0 {
+		parent.delEdge(label)
```

Comment:
Yes, in fact we *want* to sometimes do both.  If we have a parent node with two children and we delete one of them, this delEdge call will remove the node we've just deleted from the list of children and then the parent.mergeChild() call will collapse the *other* child into the parent node.